packages/ckeditor5-html-support/src/dataschema.ts

```diff
@@ -150,21 +150,31 @@ export default class DataSchema extends Plugin {
 	 * @param modelName Data schema model name.
 	 */
 	private* _getReferences( modelName: string ): Iterable<DataSchemaDefinition> {
-		const { modelSchema } = this._definitions.get( modelName )!;
-
-		if ( !modelSchema ) {
-			return;
-		}
+		const definitions = this._definitions.filter( definition => definition.model == modelName );
 
-		const inheritProperties = [ 'inheritAllFrom', 'inheritTypesFrom', 'allowWhere', 'allowContentOf', 'allowAttributesOf' ];
-
-		for ( const property of inheritProperties ) {
-			for ( const referenceName of toArray( ( modelSchema as any )[ property ] || [] ) ) {
-				const definition = this._definitions.get( referenceName );
+		for ( const { modelSchema } of definitions ) {
+			if ( !modelSchema ) {
+				continue;
+			}
 
-				if ( referenceName !== modelName && definition ) {
-					yield* this._getReferences( definition.model );
-					yield definition;
+			const inheritProperties = [
```

Comment:
This table will be unnecessarily recreated for each definition. I'd move it before the loop.